pkg/ruler/evaluator_remote.go

```diff
@@ -230,7 +233,7 @@ func (r *RemoteEvaluator) query(ctx context.Context, orgID, query string, ts tim
 			{Key: textproto.CanonicalMIMEHeaderKey("User-Agent"), Values: []string{userAgent}},
 			{Key: textproto.CanonicalMIMEHeaderKey("Content-Type"), Values: []string{mimeTypeFormPost}},
 			{Key: textproto.CanonicalMIMEHeaderKey("Content-Length"), Values: []string{strconv.Itoa(len(body))}},
-			{Key: textproto.CanonicalMIMEHeaderKey(string(httpreq.QueryTagsHTTPHeader)), Values: []string{"ruler"}},
+			{Key: textproto.CanonicalMIMEHeaderKey(string(httpreq.QueryTagsHTTPHeader)), Values: []string{"ruler,Source=ruler,Rule=" + detail.Name}},
```

Comment:
This feels quite flimsy. Can we make it more robust and add some tests?

Reply:
> This feels quite flimsy.

What I would expect is that once could set `X-Query-Tags` multiple times. Unfortunately it's not documented anywhere. I did see a set for the comma separated case. What do you think would break? What should I test?

Reply:
The things I find flimsy about this are:

1. We're concatenating a string rather than using a struct which produces the final value and is independently testable
2. There aren't any tests here which prove this works
3. Having the name alone may not be enough information for us to triangulate which rule is acting up